README.md

```diff
@@ -2,15 +2,142 @@
+make this method even more flexible by making the `name` argument optional. We
+do this by using optional, or default, arguments.
+
+## Define a Method With Optional Arguments
+
+Default arguments are easy to add, you simply assign them a default value with
+`=` ("equals") in the argument list. There's no limit to the number of arguments
+that you can make default.
+
+```ruby
+def greeting(name="Ruby programmer", language="Ruby")
+  puts "Hello, #{name}. We heard you are a great #{language} programmer."
+end
+```
+
+Let's take a look at the different ways we can call this method:
+
+```ruby
+greeting
+# > Hello, Ruby programmer. We heard you are a great Ruby programmer.
+
+greeting("Maria")
+# > Hello, Maria. We heard you are a great Ruby programmer.
+
+greeting("Steven", "Elixir")
+# > Hello, Steven. We heard you are a great Elixir programmer.
+```
+
+Now, no matter whether an argument is passed in or not, text is output and no
+errors are thrown.
+
+## Define a Method With Optional and Required Arguments
+
+It is possible to define a method that takes in both required and default
+arguments. To do so, we  place the default argument at the **end** of the
+argument list in the method definition.
+
+Take a look:
+
+```ruby
+def greeting(name, language="Ruby")
+  puts "Hello, #{name}. We heard you are a great #{language} programmer."
+end
+```
+
+Let's call our `#greeting` method with and without an explicit `language`
+argument:
+
+```ruby
+greeting("Maria", "Ember.js")
+# > Hello, Maria. We heard you are a great Ember.js programmer. 
+
+greeting("Dan")
+# > Hello, Dan. We heard you are a great Ruby programmer.
+```
+
+It works! Why must we place the default argument at the end of the argument
+list?
+
+Let's take a look at what would happen if we didn't:
+
+```ruby
+def greeting(language="Ruby", name)
+  puts "Hello, #{name}. We heard you are a great #{language} programmer."
+end
+```
+
+Now, what happens when we try to call our method without an explicit `language`
+argument?
+
+```ruby
+greeting("Maria")
+```
+You might expect it to break. Or you might expect it to think that the
+`language` variable is being set equal to `"Maria"` in this method call. 
+
+Neither of those things will happen. The method will work as we intended because
+Ruby is smart and has a few tricks up its sleeve to help determine what method
+arguments are being used where in a method's body. 
+
+However, defining the default argument first is confusing. We can understand
+this from our very reasonable expectations that the above method invocation
+would break. For this reason, it is conventional to place any default arguments
+at the end of an argument list when defining a method that takes in both
+required and default arguments. 
 
-## SWBAT 2
 
 ## Conclusion
 
+Method arguments, both required and optional, make methods powerfully abstract
+and dynamic machines that are easy to build yet very flexible and adaptable to
```

Comment:
replace and => ',

We're not calling methods "machines" elsswhere.....i know the legacy curriculum likes this metaphor. 

I think this para is weak. I think you can reduce it to:

"Method arguments, both required and optional, make methods abstract so that we can cover a wide variety of usages while keeping our code DRY."